### DISCUSSION TITLE
WiFi Client: choose available AP in MultiAP-Network

### DISCUSSION THREAD

**Discussion by poldy42** [RaspAP/raspap-webgui#1340] [Ideas]
Hi all,

I use Raspberry Pi with two USB WLAN Adapter since some years to bring external WLAN into our CamperVan. On my search for a more convenient solution instead of doing all in bash I found your project and started to use it - very well done, looks perfect for all my needs. Except one point, which is maybe possible, but I am not able to find it in RaspAP. 

Many camping sites offers WLAN to their guests. Normally the have a bunch of APs distributed over the complete area, sending same SSID on different channels. But owner of these camping sites are not really network experts, they have more important topics to take care of. So it can be, that for some reason a AP is isolated from rest of network. Or not serving IP via DHCP. Or or or ... found a lot of reasons why it was not possible to use the nearest AP. But often it is a solution to connect to an AP, which hat lower signal but stable connection.

On command line I normally use to "iwlist wlan1 scanning|grep -B 3 '<WiFiSSID>'" to find all APs for the offered network and then connect directly to APs using their FRQ "iw wlan1 connect -w '<WiFiSSID>' 2432".

Can I do something similar in RaspAP -and if not: is this worth to open a feature request ?

Many thanks in advance - Michael

**Comment by billz** [comment]
Interesting use case. RaspAP uses wpa_cli, rather than iw to scan and connect to APs, eg.:

https://github.com/RaspAP/raspap-webgui/blob/c971c5c9bcd8c41814fe95202393b2a63f37f766/includes/wifi_functions.php#L66

The output of wpa_cli's scan is less detailed than iwlist, but does include frequency, signal level, etc.

Which criteria is most useful for your case when filtering available APs?

**Comment by poldy42** [comment]
Hi,

output of wpa_cli seems also to have all needed criteria - SSID, Signal strength and MAC of AP.
Reason why I choosed frequency for connection with iw was that iw is not able to connect on MAC, only on frequency. Maybe this is possible with wpa_cli ?

If I could make a proposal: have a list of available SSIDs, sorted (descending) by SSID and signal strength and choose by myself why AP to connect with. Maybe on a advanced sub page of WiFi Client. Unfortunately my programmers knowledge is ... don't ask. 

cheers - Michael

**Reply by billz** [reply]
Hi Michael,

I see what you mean. The WiFi client UI simply returns scan results directly from wpa_cli. Adding a default sort order as you describe (SSID + RSSI) would be fairly trivial to implement.

Thanks for the input. Cheers

**Reply by poldy42** [reply]
Good morning Bill,
just a quick question about my proposal - have you already been able to discuss/decide whether and how this can be integrated ?
Cheers - Michael

**Reply by billz** [reply]
Hi again Michael,
We've made some updates to the WiFi client UI, including sorting available APs similar to the method you describe.

<img width="800" alt="image" src="https://github.com/RaspAP/raspap-webgui/assets/229399/696be3f6-fb08-4448-b344-8ff7d40dc8e9">

Not sure if this fulfills your needs precisely. One of our developers who contributes to the Insiders edition also uses RaspAP for his camper van, and has integrated extra functionality for this purpose (mainly centered around support for [network devices](https://docs.raspap.com/net-devices/)).

If you'd like to evaluate Insiders I'd be happy to give you access to the private repo.

**Comment by poldy42** [comment]
Good morning Bill,

sorry for late reply, too much work, too less time.

I would be happy to try out the new function and give the developer feedback! 

However, I won't be back home until next Wednesday and could test.


Thanks in advance and have a nice weekend - Michael
